src/img2pdf.py

```diff
@@ -347,21 +376,80 @@ def valid_date(string):
+        "letter"      : "612x612^",     # "612x792",
+        "arche"       : "2592x2592^",   # "2592x3456",
+        "archd"       : "1728x1728^",   # "1728x2592",
+        "archc"       : "1296x1296^",   # "1296x1728",
+        "archb"       : "864x864^",     # "864x1296",
+        "archa"       : "648x648^",     # "648x864",
+        "a0"          : "2380x2380^",   # "2380x3368",
+        "a1"          : "1684x1684^",   # "1684x2380",
+        "a2"          : "1190x1190^",   # "1190x1684",
+        "a3"          : "842x842^",     # "842x1190",
+        "a4"          : "595x595^",     # "595x842",
+        "a5"          : "421x421^",     # "421x595",
+        "a6"          : "297x297^",     # "297x421",
+        "a7"          : "210x210^",     # "210x297",
+        "a8"          : "148x148^",     # "148x210",
+        "a9"          : "105x105^",     # "105x148",
+        "a10"         : "74x74^",       # "74x105",
+        "b0"          : "2836x2836^",   # "2836x4008",
+        "b1"          : "2004x2004^",   # "2004x2836",
+        "b2"          : "1418x1418^",   # "1418x2004",
+        "b3"          : "1002x1002^",   # "1002x1418",
+        "b4"          : "709x709^",     # "709x1002",
+        "b5"          : "501x501^",     # "501x709",
+        "c0"          : "2600x2600^",   # "2600x3677",
+        "c1"          : "1837x1837^",   # "1837x2600",
+        "c2"          : "1298x1298^",   # "1298x1837",
+        "c3"          : "918x918^",     # "918x1298",
+        "c4"          : "649x649^",     # "649x918",
+        "c5"          : "459x459^",     # "459x649",
+        "c6"          : "323x323^",     # "323x459",
+        "flsa"        : "612x612^",     # "612x936",
+        "flse"        : "612x612^",     # "612x936",
+        "halfletter"  : "396x396^",     # "396x612",
+        "tabloid"     : "792x792^",     # "792x1224",
+        "statement"   : "396x396^",     # "396x612",
+        "executive"   : "540x540^",     # "540x720",
+        "folio"       : "612x612^",     # "612x936",
+        "quarto"      : "610x610^",     # "610x780"
+    }
+
+    try:
+        string = papersizes[string.lower()]
+    except KeyError, e:
+        pass
+
+    pattern = re.compile('([0-9]*\.?[0-9]*)([a-z]*)x([0-9]*\.?[0-9]*)([a-z]*)([^0-9a-z]*)')
+    tokens = pattern.match(string)
+
+    # tokens.group(0) == should match entire input string
+    # tokens.group(1) == x (width)
+    # tokens.group(2) == units -- not currently used; defaults to points
+    # tokens.group(3) == y (height)
+    # tokens.group(4) == units -- not currently used; defaults to points
+    # tokens.group(5) == extra options
+
+    if tokens.group(0) == '':
         msg = "input size needs to be of the format Ax, xB or AxB with A and B being integers"
         raise argparse.ArgumentTypeError(msg)
-    x = tokens[0]
-    y = tokens[1]
-    if x == '':
+
+    try:
+        x = float(tokens.group(1))
+    except ValueError, e:
         x = None
```

Comment:
The regex also matches "." and the empty string.  In both cases, we really do want x or y to be assigned to None.